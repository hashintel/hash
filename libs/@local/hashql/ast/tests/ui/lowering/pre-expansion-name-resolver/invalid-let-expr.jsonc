//@ run: pass
//@ description: Test let expression with non-path target is not treated specially
["let", ["+", "a", "b"], "c", "body"]
