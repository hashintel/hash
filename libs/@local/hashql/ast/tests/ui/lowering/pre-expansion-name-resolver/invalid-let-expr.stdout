Expr#4294967040@25
  ExprKind (Call)
    CallExpr#4294967040@25
      Expr#4294967040@3
        ExprKind (Path)
          Path#4294967040@3 (rooted: true)
            PathSegment#4294967040@2 (name: kernel)
            PathSegment#4294967040@2 (name: special_form)
            PathSegment#4294967040@2 (name: let)
      Argument#4294967040@16
        Expr#4294967040@16
          ExprKind (Call)
            CallExpr#4294967040@16
              Expr#4294967040@7
                ExprKind (Path)
                  Path#4294967040@7 (rooted: true)
                    PathSegment#4294967040@6 (name: math)
                    PathSegment#4294967040@6 (name: add)
              Argument#4294967040@11
                Expr#4294967040@11
                  ExprKind (Path)
                    Path#4294967040@11 (rooted: false)
                      PathSegment#4294967040@10 (name: a)
              Argument#4294967040@15
                Expr#4294967040@15
                  ExprKind (Path)
                    Path#4294967040@15 (rooted: false)
                      PathSegment#4294967040@14 (name: b)
      Argument#4294967040@20
        Expr#4294967040@20
          ExprKind (Path)
            Path#4294967040@20 (rooted: false)
              PathSegment#4294967040@19 (name: c)
      Argument#4294967040@24
        Expr#4294967040@24
          ExprKind (Path)
            Path#4294967040@24 (rooted: false)
              PathSegment#4294967040@23 (name: body)
